src/main/java/com/google/sps/servlets/PostQuestionServlet.java

```diff
@@ -0,0 +1,122 @@
+  /** 
+   * This method will execute the query to post a question to the database.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String title = request.getParameter("question-title");
+    String body = request.getParameter("question-body");
+    int asker_id = Utility.getUserId();
+
+    // First we query the number of questions that exist so that we can update the
+    // QuestionFollower table as well.
+    try {
+      Connection connection = DriverManager
+          .getConnection(Utility.SQL_LOCAL_URL, Utility.SQL_LOCAL_USER, Utility.SQL_LOCAL_PASSWORD);
+      insertNewQuestion(connection, title, body, asker_id);
+      int newQuestionId = getNewQuestionId(connection);
+      insertNewFollower(connection, newQuestionId, asker_id);
+    } 
+    catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostQuestionServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+    response.sendRedirect("/");
+  }
+
+  /** 
+   * Inserts a question into the database.
+   */
+  private void insertNewQuestion(Connection connection, String title, String body, int asker_id) {
+    try {
+      String insertQuestionQuery = "INSERT INTO Question(title, body, asker_id, date_time) "
+          + "VALUES (?,?,?,NOW())";
+      PreparedStatement questionStatement = connection.prepareStatement(insertQuestionQuery);
+      questionStatement.setString(Constants.QUESTION_INSERT_TITLE_COLUMN, title);
+      questionStatement.setString(Constants.QUESTION_INSERT_BODY_COLUMN, body);
+      questionStatement.setInt(Constants.QUESTION_INSERT_ASKERID_COLUMN, asker_id);
+      questionStatement.executeUpdate();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostQuestionServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
+
+  /** 
+   * Gets the ID from the last question posted.
+   */
+  private int getNewQuestionId(Connection connection) {
+    int id = -1;
+    try {
+      String maxIdQuery = "SELECT MAX(id) FROM Question;";
+      PreparedStatement maxIdStatement = connection.prepareStatement(maxIdQuery);
+      ResultSet queryResult = maxIdStatement.executeQuery();
+      queryResult.next();
+      id = queryResult.getInt(Constants.QUESTION_FETCH_MAXID_COLUMN);
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostQuestionServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+
+    return id;
+  }
+
+  /** 
+   * Makes the author of the recently added question a follower of said question.
+   */
```

Comment:
This feels like it can be reused for 'following' questions in general, consider moving this to Utility for that reason. Could be done in a follow-up PR (in which case, add a TODO).

Reply:
done